gate-oauth2/src/main/java/com/netflix/spinnaker/gate/security/oauth2/provider/GithubProviderTokenServices.java

```diff
@@ -17,32 +17,36 @@
 
 import java.util.List;
 import java.util.Map;
+import lombok.Data;
 import lombok.extern.slf4j.Slf4j;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
-import org.springframework.boot.autoconfigure.security.oauth2.resource.ResourceServerProperties;
 import org.springframework.boot.context.properties.ConfigurationProperties;
-import org.springframework.security.oauth2.client.OAuth2RestOperations;
-import org.springframework.security.oauth2.client.OAuth2RestTemplate;
-import org.springframework.security.oauth2.client.resource.BaseOAuth2ProtectedResourceDetails;
-import org.springframework.security.oauth2.common.DefaultOAuth2AccessToken;
-import org.springframework.security.oauth2.common.OAuth2AccessToken;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
 import org.springframework.stereotype.Component;
+import org.springframework.web.client.RestTemplate;
 
+/**
+ * This class implements {@link SpinnakerProviderTokenServices} to verify if a user meets the
+ * provider-specific authentication requirements for GitHub.
+ *
+ * <p>It checks if a user is a member of a required GitHub organization before granting access.
+ */
 @Slf4j
 @Component
-@ConditionalOnProperty(name = "security.oauth2.provider-requirements.type", havingValue = "github")
+@ConditionalOnProperty(name = "spring.security.oauth2.client.registration.github.client-id")
```

Comment:
If the tests in OAuth2Test passed without this class existing, we need some more test coverage.

Reply:
Updated the test class
https://github.com/spinnaker/gate/pull/1887/commits/2b67a513a18c7121e81510a3ed10573acb5603db